src/main/java/com/ibm/domain/scanning/ScanAggregate.java

```diff
@@ -35,45 +40,70 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public final class ScanAggregate extends AggregateRoot<ScanId> {
-    private static final Logger LOGGER = LoggerFactory.getLogger(ScanAggregate.class);
+    // private static final Logger LOGGER = LoggerFactory.getLogger(ScanAggregate.class);
 
-    @Nonnull private final ScanRequest scanRequest;
+    @Nonnull private ResolvedScanRequest request;
     @Nullable private Commit commit;
     @Nullable private Map<Language, LanguageScan> languageScans;
 
     private ScanAggregate(@Nonnull final ScanId id, @Nonnull final ScanRequest scanRequest) {
         super(id, new ArrayList<>());
-        this.scanRequest = scanRequest;
+        this.request = new ResolvedScanRequest(scanRequest);
+        if (request.scanUrl.isPurl()) {
+            try {
+                PackageURL purl = new PackageURL(request.scanUrl.value());
+                if (purl.getVersion() != null) {
+                    this.request.setRevision(new Revision(purl.getVersion()));
+                }
+            } catch (MalformedPackageURLException e) {
+                // ignore: scanRequest ha been validated
+            }
+        } else {
+            request.setGitUrl(request.scanUrl);
+        }
     }
 
     private ScanAggregate(
             @Nonnull ScanId id,
             @Nonnull ScanRequest scanRequest,
+            @Nullable ScanUrl gitUrl,
```

Comment:
I think this has to be changes when applying the changes from above

Reply:
I reactivated the old GitUrl class to represent the gitUrl in ScanAggregate which is passed in here.